### ISSUE TITLE
Peers with ip v4

### ISSUE THREAD

**Issue by Fox-khb** [donaldzou/WGDashboard#330]
**Describe The Problem**
After updating to v. 4.0.2 I can only add peers with IP v 6. Even if I manually enter IP v 4 it says “This IP is not available:”

**OS Information:**
 - OS: Ubuntu 22.04
 - Python Version: v3.12

**Comment by DaanSelen** [comment]
Can you perhaps output the contents of the corresponding wireguard configuration file (REMOVE THE KEYS and other sensitive details) and show a screenshot of the dashboard with the error?

**Comment by Fox-khb** [comment]
Hello.
Here is the content of wg0.conf

```
[Interface]
Address = 10.66.66.1/24
Address = fd42:42:42::1/64
PostUp = iptables -I INPUT -p udp --dport 62770 -j ACCEPT
PostUp = iptables -I FORWARD -i eth0 -o wg0 -j ACCEPT
PostUp = iptables -I FORWARD -i wg0 -j ACCEPT
PostUp = iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE
PostUp = ip6tables -I FORWARD -i wg0 -j ACCEPT
PostUp = ip6tables -t nat -A POSTROUTING -o eth0 -j MASQUERADE
PostDown = iptables -D INPUT -p udp --dport 62770 -j ACCEPT
PostDown = iptables -D FORWARD -i eth0 -o wg0 -j ACCEPT
PostDown = iptables -D FORWARD -i wg0 -j ACCEPT
PostDown = iptables -t nat -D POSTROUTING -o eth0 -j MASQUERADE
PostDown = ip6tables -D FORWARD -i wg0 -j ACCEPT
PostDown = ip6tables -t nat -D POSTROUTING -o eth0 -j MASQUERADE
ListenPort = 62770
PrivateKey = IFHvaIUVbH7Y+U/QRuKLzho7fLaucOwpHOPmG7C0z0k=
```
![2024-08-25 15 52 01](https://github.com/user-attachments/assets/9e58d282-bd27-4b50-99be-c7738b6347b4)

**Comment by donaldzou** [comment]
Ahh yes, I spotted the bug and will update the dashboard later today :)

**Comment by donaldzou** [comment]
Hi! Please try to run `./wgd.sh update` see if you can add multiple IP now :)

**Comment by Fox-khb** [comment]
Hello.
`./wgd.sh update` didn't help. 
I solved the problem by creating a new configuration via WGDashboard specifying IP v4.

**Comment by donaldzou** [comment]
Ahh i see why.. You could adjust your configuration from:

```
[Interface]
Address = 10.66.66.1/24
Address = fd42:42:42::1/64
...
```

to

```
[Interface]
Address = 10.66.66.1/24, fd42:42:42::1/64
...
```

and this should be good :)

**Comment by DaanSelen** [comment]
> Hello. Here is the content of wg0.conf
> 
> ```
> [Interface]
> Address = 10.66.66.1/24
> Address = fd42:42:42::1/64
> PostUp = iptables -I INPUT -p udp --dport 62770 -j ACCEPT
> PostUp = iptables -I FORWARD -i eth0 -o wg0 -j ACCEPT
> PostUp = iptables -I FORWARD -i wg0 -j ACCEPT
> PostUp = iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE
> PostUp = ip6tables -I FORWARD -i wg0 -j ACCEPT
> PostUp = ip6tables -t nat -A POSTROUTING -o eth0 -j MASQUERADE
> PostDown = iptables -D INPUT -p udp --dport 62770 -j ACCEPT
> PostDown = iptables -D FORWARD -i eth0 -o wg0 -j ACCEPT
> PostDown = iptables -D FORWARD -i wg0 -j ACCEPT
> PostDown = iptables -t nat -D POSTROUTING -o eth0 -j MASQUERADE
> PostDown = ip6tables -D FORWARD -i wg0 -j ACCEPT
> PostDown = ip6tables -t nat -D POSTROUTING -o eth0 -j MASQUERADE
> ListenPort = 62770
> PrivateKey = IFHvaIUVbH7Y+U/QRuKLz=
> ```
> 
> ![2024-08-25 15 52 01](https://private-user-images.githubusercontent.com/146837636/361189060-9e58d282-bd27-4b50-99be-c7738b6347b4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk4MDU1NzIsIm5iZiI6MTcyOTgwNTI3MiwicGF0aCI6Ii8xNDY4Mzc2MzYvMzYxMTg5MDYwLTllNThkMjgyLWJkMjctNGI1MC05OWJlLWM3NzM4YjYzNDdiNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyNFQyMTI3NTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NWRjYjViNjVhODQ0NDNhZGVkMzRkODhmNWNjOWY0ZjE2ZTdkODk2ZmY4M2IzMWM4NjFhNWI1NDY2YzE1OTIxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.qRACJSB7JEvp2Yo_UFHrxURvwgkOFYu5jyeFgBi537Y)

Also, please change your private keys if you still use these. These have been exposed here on Github and are no longer safe.

**Comment by nikita-kapkovich** [comment]
I observe the same problem! I don't use IPv6! How to solve the problem? Updated from 3.0.6
![Снимок экрана 2024-10-28 в 13 37 55](https://github.com/user-attachments/assets/c4256460-cf7d-49d4-9af5-907a3fdb897c)

**Comment by DaanSelen** [comment]
@nikita-kapkovich does this happen in v4.1?

**Comment by nikita-kapkovich** [comment]
> does this happen in v4.1?

Yes, there are no changes

**Comment by donaldzou** [comment]
> > does this happen in v4.1?
> 
> Yes, there are no changes

Hi, the reason is WGDashboard can't allocate more IP address due to your configuration. Your configuration's address is `10.10.10.1/32`, which means there's only **1** IP address is available (`10.10.10.1`), if you're interested why, you can go over this great [article](https://www.auvik.com/franklyit/blog/what-is-subnet-mask/).

In your case, you'll need a bigger subnet for your configuration. What you can do is to change your `Address` under `[Interface]` to something like `10.10.10.1/24`, so now WGDashboard can allocate 254 IP addresses (`10.10.10.2` - `10.10.10.254`)

Let me know if this works :)

**Comment by nikita-kapkovich** [comment]
> > > does this happen in v4.1?
> > 
> > 
> > Yes, there are no changes
> 
> Hi, the reason is WGDashboard can't allocate more IP address due to your configuration. Your configuration's address is `10.10.10.1/32`, which means there's only **1** IP address is available (`10.10.10.1`), if you're interested why, you can go over this great [article](https://www.auvik.com/franklyit/blog/what-is-subnet-mask/).
> 
> In your case, you'll need a bigger subnet for your configuration. What you can do is to change your `Address` under `[Interface]` to something like `10.10.10.1/24`, so now WGDashboard can allocate 254 IP addresses (`10.10.10.2` - `10.10.10.254`)
> 
> Let me know if this works :)

Thanks! The issue has been resolved!
